validation/validate_dc2_run2.1i_object_table.ipynb

```diff
@@ -234,7 +256,42 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "df = pd.DataFrame(cat.get_quantities(columns, filters=[subsample] + quality_cuts))"
+    "def get_pg(schema_name, columns, sampling_factor, quality_cut, tract=None):\n",
+    "    import psycopg2\n",
+    "    dbname = 'desc_dc2_drp'\n",
+    "    dbuser = 'desc_dc2_drp_user'\n",
+    "    dbhost = 'nerscdb03.nersc.gov'\n",
+    "    dbconfig = {'dbname' : dbname, 'user' : dbuser, 'host' : dbhost}\n",
+    "    dbconn = psycopg2.connect(**dbconfig)  \n",
+    "    \n",
+    "    to_select = ','.join(columns)   \n",
+    "    cuts = []\n",
+    "    if tract is not None:\n",
+    "        tract_cut = f' tractsearch(objectid, {tract}) '\n",
+    "        cuts = [tract_cut]\n",
+    "\n",
+    "    if sampling_factor > 1:\n",
+    "        arbitrary_last_digit = '2' \n",
+    "        remainder = int(math.ceil(math.log10(sampling_factor)) * arbitrary_last_digit)\n",
+    "        remainder = remainder % sampling_factor\n",
+    "        #sample_clause = ' (objectid % {}) = {} '.format(sampling_factor, arbitrary_last_digit)\n",
+    "        sample_clause = f' (objectid % {sampling_factor}) = {arbitrary_last_digit} '\n",
+    "        cuts.append(sample_clause)  \n",
+    "        \n",
+    "    if quality_cut is not None: cuts.append(quality_cut)\n",
+    "    all_cuts = ' and '.join(cuts)\n",
+    "    \n",
+    "    where = ' where '\n",
+    "    where  += all_cuts \n",
+    "\n",
+    "    q = f'SELECT {to_select} FROM {schema_name}.dpdd {where}'\n",
```

Comment:
One can supply parameters to be substituted to the psycopg2 execute command.  I looked into it while writing up a minimal SQL primer (which is in a different pull request) and decided not to recommend it there because
* it involves yet another set of quoting/escape rules to keep track of
* the account people use to access the Postgres database is read-only.